server/schedule/mockcluster.go

```diff
@@ -376,55 +376,61 @@ func (mc *MockCluster) newMockRegionInfo(regionID uint64, leaderID uint64, follo
 	return core.NewRegionInfo(region, leader)
 }
 
-// ApplyOperator mocks apply oeprator.
+// ApplyOperatorStep mocks apply operator step.
+func (mc *MockCluster) ApplyOperatorStep(region *core.RegionInfo, op *Operator) *core.RegionInfo {
+	if step := op.Check(region); step != nil {
+		switch s := step.(type) {
+		case TransferLeader:
+			region = region.Clone(core.WithLeader(region.GetStorePeer(s.ToStore)))
+		case AddPeer:
+			if region.GetStorePeer(s.ToStore) != nil {
+				panic("Add peer that exists")
+			}
+			peer := &metapb.Peer{
+				Id:      s.PeerID,
+				StoreId: s.ToStore,
+			}
+			region = region.Clone(core.WithAddPeer(peer))
+		case RemovePeer:
+			if region.GetStorePeer(s.FromStore) == nil {
+				panic("Remove peer that doesn't exist")
+			}
+			if region.GetLeader().GetStoreId() == s.FromStore {
+				panic("Cannot remove the leader peer")
+			}
+			region = region.Clone(core.WithRemoveStorePeer(s.FromStore))
+		case AddLearner:
+			if region.GetStorePeer(s.ToStore) != nil {
+				panic("Add learner that exists")
+			}
+			peer := &metapb.Peer{
+				Id:        s.PeerID,
+				StoreId:   s.ToStore,
+				IsLearner: true,
+			}
+			region = region.Clone(core.WithAddPeer(peer))
+		case PromoteLearner:
+			if region.GetStoreLearner(s.ToStore) == nil {
+				panic("promote peer that doesn't exist")
```

Comment:
It will be better to keep the first letter consistent with other panic messages.